packages/next/src/server/typescript/utils.ts

```diff
@@ -16,13 +24,35 @@ export function init(opts: {
   ts: TypeScript
   info: tsModule.server.PluginCreateInfo
 }) {
+  const projectDir = opts.info.project.getCurrentDirectory()
   ts = opts.ts
   info = opts.info
-  const projectDir = info.project.getCurrentDirectory()
   appDirRegExp = new RegExp(
     '^' + (projectDir + '(/src)?/app').replace(/[\\/]/g, '[\\/]')
   )
-  log('Starting Next.js TypeScript plugin: ' + projectDir)
+
+  log('[next] Initializing Next.js TypeScript plugin: ' + projectDir)
+
+  const compilerOptions = info.project.getCompilerOptions()
+  const fsMap = createDefaultMapFromNodeModules(
+    compilerOptions,
+    ts,
+    join(projectDir, 'node_modules/typescript/lib')
+  )
+  const system = createFSBackedSystem(fsMap, projectDir, ts)
+
+  virtualTsEnv = createVirtualTypeScriptEnvironment(
+    system,
+    [],
+    ts,
+    compilerOptions
+  )
+
+  if (!virtualTsEnv) {
+    throw new Error('[next] Failed to create virtual TypeScript environment.')
```

Comment:
Maybe we mention that this is a bug in Next.js, and logs should be shared in a GitHub issue, so that the user doesn't think this is a problem with their code